### PR TITLE
`AddInternalAsync` refactor and explicit `RemoveAsync`

### PR DESCRIPTION
**Pull request by emanuelgaspar**
The detection of trying to `SubRemove` subcriptions when calling `AddInternalAsync` was difficult to understand due to its use of `JsonDocument.Parse(subscription.Topic)`.

This PR introduces a strongly-typed deserialization of the topic, which allows us to work on the array of subs with clean code.

New `RemoveAsync(TopicSubscription subscription)` method added to `ICryptoCompareWebsocketHandler`.
There's one from the base interface in `websockets` which takes an `IList` of subscriptions, but doesn't do what we need in `market-data`.

Also bumped some nuget/actions versions.

### CODE REVIEW

File: src/Trakx.CryptoCompare.ApiClient.Websocket.Tests/Unit/CryptoCompareWebsocketHandlerTests.cs [hunk 1]
```diff
@@ -75,4 +114,10 @@ public async Task IncomingMessage_should_map_to_correct_observer()
         topicMessage!.Symbol.Should().Be(fullVolumeMsg.Symbol);
         topicMessage!.Volume.Should().Be(fullVolumeMsg.Volume);
     }
+
+    private static TopicSubscription SetupTopicSubscription(SubscribeActions action)
+    {
+        var subscriptionString = CryptoCompareSubscriptionFactory.GetFullTopTierVolumeSubscriptionStr("test");
```
Comment (goneneren) [inline src/Trakx.CryptoCompare.ApiClient.Websocket.Tests/Unit/CryptoCompareWebsocketHandlerTests.cs:120]: [suggestion] maybe use a more meaningful string then 'test'

File: src/Trakx.CryptoCompare.ApiClient.Websocket/CryptoCompareWebsocketHandler.cs [hunk 2]
```diff
@@ -61,30 +63,38 @@ public class CryptoCompareWebsocketHandler : ClientWebsocketRedirectHandlerBase<
...
+        SendClient(subscription.Topic);
+
+        await Task.CompletedTask;
+        return true;
     }
 
-    protected override Uri ClientSocketUri => _config.WebSocketEndpoint;
+    private static bool HasRemoveAction(TopicSubscription subscription)
+    {
+        var remove = SubscribeActions.SubRemove.ToString();
```
Comment (goneneren) [inline src/Trakx.CryptoCompare.ApiClient.Websocket/CryptoCompareWebsocketHandler.cs:97]: Can you add some summary comments to methods and maybe choose a different name for the variable 'remove' I have difficulty to understand what this method does

File: src/Trakx.CryptoCompare.ApiClient.Websocket.Tests/Unit/CryptoCompareWebsocketHandlerTests.cs (hunk 1 above)
Comment (emanuelgaspar) [reply src/Trakx.CryptoCompare.ApiClient.Websocket.Tests/Unit/CryptoCompareWebsocketHandlerTests.cs:120]: "test" comes from the previous version. Would "testSymbol" work here?

File: src/Trakx.CryptoCompare.ApiClient.Websocket/CryptoCompareWebsocketHandler.cs [hunk 3]
```diff
@@ -61,30 +63,38 @@ public class CryptoCompareWebsocketHandler : ClientWebsocketRedirectHandlerBase<
...
+    }
+
+    public async Task<bool> RemoveAsync(TopicSubscription subscription)
+    {
+        var payload = JsonSerializer.Deserialize<CryptoCompareSubscription>(subscription.Topic);
+
+        if (payload == null) return false;
+        if (payload.Action != SubscribeActions.SubRemove) return false;
+
+        foreach (var item in payload.Subs)
```
Comment (goneneren) [inline src/Trakx.CryptoCompare.ApiClient.Websocket/CryptoCompareWebsocketHandler.cs:80]: How about:

```
foreach (var payloadSub in payload.Subs)
        {
            var subsToRemove = Subscriptions.Where(t => t.Topic.ContainsIgnoreCase(payloadSub)).ToList();
            foreach (var sub in subsToRemove )
            {
                Subscriptions.Remove(sub);
            }
        }
```

Comment (goneneren) [reply src/Trakx.CryptoCompare.ApiClient.Websocket/CryptoCompareWebsocketHandler.cs:80]: renaming of the variables

**Review by goneneren — CHANGES_REQUESTED** [review]

File: src/Trakx.CryptoCompare.ApiClient.Websocket.Tests/Unit/CryptoCompareWebsocketHandlerTests.cs (hunk 1 above)
Comment (emanuelgaspar) [reply src/Trakx.CryptoCompare.ApiClient.Websocket.Tests/Unit/CryptoCompareWebsocketHandlerTests.cs]: Done via const.

File: src/Trakx.CryptoCompare.ApiClient.Websocket/CryptoCompareWebsocketHandler.cs (hunk 2 above)
Comment (emanuelgaspar) [reply src/Trakx.CryptoCompare.ApiClient.Websocket/CryptoCompareWebsocketHandler.cs]: Sure, done.

**Review by goneneren — APPROVED** [review]

**Pull request merged**